### PR TITLE
Convert Comscore to TypeScript

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
convert comscore and comscore.spec to TypeScript

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
makes commercial modules easier to maintain and work with

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/comscore.ts [hunk 1]
```diff
@@ -39,35 +34,41 @@ const initOnConsent = () => {
...
-			})
-			.catch((e) => {
-				log('commercial', '⚠️ Failed to execute comscore', e);
-			});
+					if (!(canRunTcfv2 || canRunAus || canRunCcpa)) {
+						return Promise.reject('No consent for comscore');
+					}
+					return initOnConsent();
+				})
+				// is there a better way?
```
Comment (zekehuntergreen) [inline static/src/javascripts/projects/commercial/modules/comscore.ts:57]: Is there?

Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/comscore.ts:57]: I think this logic could move to [`getConsentFor`](https://github.com/guardian/consent-management-platform/blob/main/src/getConsentFor.ts).

File: static/src/javascripts/projects/commercial/modules/comscore.ts [hunk 2]
```diff
@@ -39,35 +34,41 @@ const initOnConsent = () => {
  * Initialise comscore, industry-wide audience tracking
  * https://www.comscore.com/About
  */
-const setupComscore = () => {
+const setupComscore = (): Promise<void> => {
 	if (commercialFeatures.comscore) {
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/comscore.ts:38]: Maybe an early return here would help with the deep indentation

**Review by mxdvl — COMMENTED** [review]
I think we should strive for using one type of logic: chained `then`s or `if` blocks with awaits. Currently there’s a mix of conditionals and rejecting promises with the catch block.

File: static/src/javascripts/projects/commercial/modules/comscore.spec.ts [hunk 3]
```diff
@@ -76,14 +92,14 @@ jest.mock('../../common/modules/commercial/commercial-features', () => ({
 describe('setupComscore', () => {
 	it('should do nothing if the comscore is disabled in commercial features', () => {
 		commercialFeatures.comscore = false;
-		setupComscore();
+		void setupComscore();
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/comscore.spec.ts:95]: I think you’d need to either `await` this, or return it and chain the expect check inside the `then`

Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/comscore.spec.ts:95]: I think if this returns a promise then it should be `await`-ed and the test converted to `async`?

File: static/src/javascripts/projects/commercial/modules/comscore.spec.ts [hunk 4]
```diff
@@ -76,14 +92,14 @@ jest.mock('../../common/modules/commercial/commercial-features', () => ({
...
 
 		expect(onConsentChange).not.toBeCalled();
 	});
 
 	it('should register a callback with onConsentChange if enabled in commercial features', () => {
 		commercialFeatures.comscore = true;
-		setupComscore();
+		void setupComscore();
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/comscore.spec.ts:102]: Ditto wrt `await`

File: static/src/javascripts/projects/commercial/modules/comscore.ts (hunk 1 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/comscore.ts:57]: Adding a todo

File: static/src/javascripts/projects/commercial/modules/comscore.ts (hunk 2 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/comscore.ts]: Good call

**Review by mxdvl — APPROVED** [review]

**Pull request merged**